virtool/indexes/api.py

```diff
@@ -236,3 +239,44 @@ async def find_history(req):
     )
 
     return json_response(data)
+
+
+async def reset_history(history: Collection, index_id: str):
+    """
+    Set the index.id and index.version fields with the given index id to 'unbuilt'.
+
+    :param history: The history collection of the database.
+    :param index_id: The ID of the index which failed to build
+    """
+    query = {
+        "_id": {
+            "$in": await history.distinct("_id", {"index.id": index_id})
+        }
+    }
+
+    return await history.update_many(query, {
+        "$set": {
+            "index": {
+                "id": "unbuilt",
+                "version": "unbuilt"
+            }
+        }
+    })
+
+
+@routes.delete("/api/indexes/{index_id}", jobs_only=True)
+async def delete_index(req: aiohttp.web.Request):
+    """Delete the index with the given id and reset history relating to that index."""
+    index_id = req.match_info["index_id"]
+    db = req.app["db"]
+    indexes: Collection = db.indexes
```

Comment:
I think you are trying to improve type hinting by declaring `indexes` here instead of calling `db.indexes.delete_one`. I'd prefer foregoing nice type hinting in favour of avoiding unnecessary variable declarations.

We can improve the implementation of `virtool.db.core` once the workflow work is done to solve the typing problem.